packages/xod-client-electron/src/upload/components/PopupUploadConfig.jsx

```diff
@@ -273,7 +274,10 @@ class PopupUploadConfig extends React.Component {
             onChange={this.onCloudCompilationChanged}
           />
           <label htmlFor="compileInCloud">
-            Compile in the cloud (experimental)
+            Compile in the cloud&nbsp;
+            {compileLimit
+              ? <span>({compileLimit.left} left)</span>
+              : <span>(currently offline)</span>}
```

Comment:
Probably, we should also disable this option in that case

Reply:
@evgenykochetkov 

```typescript
compileLimit: CompilationLimit | null
```
where `null` signals the absence of the compilation quota for a user. Basically, it's not about networking.

Reply:
But the message is about networking... 😬

Reply:
+ we should disable this option if no quota left

Reply:
@brusherru https://github.com/xodio/xod/issues/829

> If the request fails to complete successfully (no internet, service down), I see “[ ] Compile in the cloud (currently offline)”.

Reply:
From a user point of view, we should disable if offline or quota exhaust. If it is trivial, please, do it.

Reply:
So, if service is offline — can we try to compile in the cloud? :)
I think, no, cause it meaningless.
But we should retry to connect, to turn it on when service becomes available again.